server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/PartitionFilesSorter.java

```diff
@@ -58,22 +71,40 @@ public class PartitionFilesSorter {
   protected final long sortTimeout;
   protected final long fetchChunkSize;
   protected final long reserveMemoryForSingleSort;
+  private long partitionSorterShutdownAwaitTime;
+  private DB sortedFilesDb;
+
   protected final AbstractSource source;
 
   private final ExecutorService fileSorterExecutors = ThreadUtils.newDaemonCachedThreadPool(
     "worker-file-sorter-execute", Math.max(Runtime.getRuntime().availableProcessors(), 8), 120);
   private final Thread fileSorterSchedulerThread;
 
-  PartitionFilesSorter(MemoryTracker memoryTracker, long sortTimeOut, long fetchChunkSize,
-    long reserveMemoryForSingleSort, AbstractSource source) {
-    this.sortTimeout = sortTimeOut;
-    this.fetchChunkSize = fetchChunkSize;
-    this.reserveMemoryForSingleSort = reserveMemoryForSingleSort;
+  PartitionFilesSorter(MemoryTracker memoryTracker, RssConf conf, AbstractSource source) {
+    this.sortTimeout = RssConf.partitionSortTimeout(conf);
+    this.fetchChunkSize = RssConf.workerFetchChunkSize(conf);
+    this.reserveMemoryForSingleSort =  RssConf.memoryReservedForSingleSort(conf);
+    this.partitionSorterShutdownAwaitTime = RssConf.partitionSorterCloseAwaitTimeMs(conf);
     this.source = source;
+    // ShuffleClient only can fetch shuffle data from a restarted worker only
+    // when the worker's fetching port is stable and enables graceful shutdown.
+    if (RssConf.workerGracefulShutdown(conf)) {
+      try {
+        String recoverPath = RssConf.workerRecoverPath(conf);
+        this.recoverFile = new File(recoverPath, RECOVERY_SORTED_FILES_FILE_NAME);
+        this.sortedFilesDb = LevelDBProvider.initLevelDB(recoverFile, CURRENT_VERSION);
+        reloadSortedShuffleFiles(this.sortedFilesDb);
```

Comment:
I think we'd better to clear the db after reloading to avoid residual

Reply:
Done